service/history/transfer_queue_active_task_executor.go

```diff
@@ -103,6 +103,7 @@ func newTransferQueueActiveTaskExecutor(
 			shard,
 			workflowCache,
 			logger,
+			false,
```

Comment:
this is for the bad binary case right? I think it should behave the same as the reset api.

Reply:
There's also a pending children check in `closeTransactionHandleWorkflowResetTask` you might want to change.

Reply:
> this is for the bad binary case right? I think it should behave the same as the reset api.

Done.

> There's also a pending children check in closeTransactionHandleWorkflowResetTask you might want to change.

Didn't know about that. Will include it.

Reply:
Looked into `closeTransactionHandleWorkflowResetTask`. This is a method on mutable state and wiring this config here seems to touch a lot of code. Am I missing something? How important is this?

Reply:
Synced offline. let's remove the check in closeTransactionHandleWorkflowResetTask.